_automating-workflows/workflow-tutorial.md

```diff
@@ -0,0 +1,613 @@
+
+If you define `previous_node_inputs`, then defining edges is optional.
+{: .note}
+
+## 2. Use the deployed model for inference
+
+A CoT agent can use the deployed model in a tool using the [ML Commons Agent Framework](Link TBD). This step doesn’t strictly correspond to an API but represents a component of the body that the Register Agent API requires. This simplifies the register request and allows reuse of the same tool in multiple agents.
+
+### Step `math_tool`
+
+You can configure other tools that the CoT agent can use. For example, you can configure a math tool as follows. The math tool does not depend on any previous steps:
+
+```yaml
+- id: math_tool
+  type: create_tool
+  user_inputs:
+    name: MathTool
+    type: MathTool
+    description: A general tool to calculate any math problem. The action input
+      must be a valid math expression, like 2+3
+    parameters:
+      max_iteration: 5
+```
+
+### Step `sub_agent`
+
+To use the math tool in the agent configuration, specify it as one of the tools in the `previous_node_inputs` field of the agent. You can add other tools to `previous_node_inputs` as necessary. The agent also needs a large language model (LLM) to reason with the tools, which is defined by the `llm.model_id` field. This example assumes the `model_id` from the `deploy_model_3` step will be used. However, if another model is already deployed, the `model_id` of that previously deployed model could be included in the `user_inputs` field instead:
+
+```yaml
+- id: sub_agent
+  type: register_agent
+  previous_node_inputs:
+    # When llm.model_id is not present this can be used as a fallback value
+    deploy-model-3: model_id
+    math_tool: tools
+  user_inputs:
+    name: Sub Agent
+    type: conversational
+    description: this is a test agent
+    parameters:
+      hello: world
+    llm.parameters:
+      max_iteration: '5'
+      stop_when_no_tool_found: 'true'
+    memory:
+      type: conversation_index
+    app_type: chatbot
+```
+
+OpenSearch will automatically create the following edges so that the agent can retrieve these fields from the previous node: 
+
+```yaml
+- source: math_tool
+  dest: sub_agent
+- source: deploy_model_3
+  dest: sub_agent
+```
+
+### Step `agent_tool`
+
+You can use agents as tools for another agent. Registering an agent produces an `agent_id` in the output. The following step defines a tool which uses the `agent_id` from the previous step:
+
+```yaml
+- id: agent_tool
+  type: create_tool
+  previous_node_inputs:
+    sub_agent: agent_id
+  user_inputs:
+    name: AgentTool
+    type: AgentTool
+    description: Agent Tool
+    parameters:
+      max_iteration: 5
+```
+
+OpenSearch automatically creates an edge connection because this step specifies the `previous_node_input`:
+
+```yaml
+- source: sub_agent
+  dest: agent_tool
+```
+
+### Step `ml_model_tool`
+
+A tool may reference an ML model. This example gets the required `model_id` from the model deployed in a previous step:
```

Comment:
"a" model? "the" is correct if only one model has ever been deployed in a previous step (which I can't remember 😄).